### DISCUSSION TITLE
Some thoughts on BOM import

### DISCUSSION THREAD

**Discussion by SergeoLacruz** [inventree/InvenTree#4739] [Ideas]
Hi,
We get the BOM out of our schematic tools and can import those into InvenTree. Since the reference limit has changed there is no problem an more :-) It works great even with BOMs of more than 200 codes. 
Unfortunately the design process is not always that clean. Design work is still done when other people check stock and start to order.
Problem:
I have uploaded a BOM and the material team already did stock allocations on some parts
The design team provides a new revision of the BOM
I upload the BOM in InvenTree and have two options

- Clear existing BOM before upload: the resultin BOM is OK but all stock allocations are gone
- Do not clear: New parts are added, removed parts are still in, quantity changes are silently ignored. 

Hmm....
In an ideal case I would like to have a window telling
- Following parts will be added...
- Following parts will be deleted...
- Following parts will be changed...

This can be put inside a plugin but I think it makes sense to have in in the main. 

Michael

**Comment by SchrodingersGat** [comment]
@SergeoLacruz please keep in mind that we have a (very "long running") plan for a major overhaul of the importer tool so provide many imrpovements:

https://github.com/inventree/InvenTree/issues/3101

The new import tool will be much more "interactive" and allow for more complex (and hopefully, intuitive) user interaction.

Note that this feature has been "Planned" for a long time as we develop this software in our "free" time. If your company would benefit from this appearing sooner, please LMK as you could potentially sponsor development quicker.

**Reply by SergeoLacruz** [reply]
Hi,
Thanks for the info. It is good to know that there is something on the roadmap. I did not know that. It seems quite big. Might be I can find some sponsoring, but this will take some time...

**Comment by SergeoLacruz** [comment]
This is starting to get painful, We are thinking about putting this function into a plugin which should be straight forward. Does this make sense or is 3101 short before release?

**Reply by matmair** [reply]
We are mainly working on the react refactor right now, the importer will probably take a while. So if you have time to implement this and it is important, a PR would be appreciated. Doing it in a plugin is possible but that would be fragile as refactors in the models would not raise any problems with the importer. So doing it the upstream codebase (maybe as a builtin plugin) might be best.
